skops/io/tests/test_audit.py

```diff
@@ -0,0 +1,111 @@
+import io
+import json
+from zipfile import ZipFile
+
+import pytest
+
+from skops.io import dumps, get_untrusted_types
+from skops.io._audit import audit_tree, check_type
+from skops.io._dispatch import Node, get_tree
+from skops.io._general import DictNode, dict_get_state
+from skops.io._utils import LoadContext, SaveContext, gettype
+
+
+class CustomType:
+    """A custom untrusted class."""
+
+    def __init__(self, value):
+        self.value = value
+
+
+@pytest.mark.parametrize(
+    "module_name, type_name, trusted, expected",
+    [
+        ("sklearn", "Pipeline", ["sklearn.Pipeline"], True),
+        ("sklearn", "Pipeline", ["sklearn.preprocessing.StandardScaler"], False),
+        ("sklearn", "Pipeline", True, True),
+    ],
+    ids=["list-True", "list-False", "True"],
+)
+def test_check_type(module_name, type_name, trusted, expected):
+    assert check_type(module_name, type_name, trusted) == expected
+
+
+def test_audit_tree_untrusted():
+    var = {"a": CustomType(1), 2: CustomType(2)}
+    state = dict_get_state(var, SaveContext(None, 0, {}))
+    node = DictNode(state, LoadContext(None), trusted=False)
+    with pytest.raises(
+        TypeError, match="Untrusted types found in the file: {'test_audit.CustomType'}."
+    ):
+        audit_tree(node, trusted=False)
```

Comment:
Instead of testing `audit_tree`, which, AFAICT, would never be used by the user directly, could we test `loads` with the same logic?